### PR TITLE
Mention concurrent builds limitation in "Build Process"

### PR DESCRIPTION
**Pull request by humitos**
This is the smallest amount of work to mention this.

However, I think that this whole page needs some work. I'll see if I can start doing that in another PR, but I think it worth to mention this somewhere in the meanwhile.

### CODE REVIEW

File: docs/builds.rst [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
-Files: `tasks.py`_ - `doc_builder/`_
-
-.. _tasks.py: https://github.com/readthedocs/readthedocs.org/blob/master/readthedocs/projects/tasks.py
-.. _doc_builder/: https://github.com/rtfd/readthedocs.org/tree/master/readthedocs/doc_builder
-
 Every documentation build has limited resources.
 Our current build limits are:
 
 * 15 minutes of CPU
 * 1GB of RAM memory
```
Comment (ericholscher) [inline docs/builds.rst:8]: These changed as well, no?

Comment (humitos) [reply docs/builds.rst:8]: Yes. I think the default is 15m and 3GB now. I'll update.

File: docs/builds.rst [hunk 2]
```diff
@@ -130,14 +126,14 @@ by default the image used is ``readthedocs/build:latest``,
 but you can change that using a :doc:`config-file/index`.
 
 .. note::
-   
+
```
Comment (ericholscher) [inline docs/builds.rst:129]: Does this not break?

Comment (ericholscher) [reply docs/builds.rst:129]: Guess not, huh.

**Pull request merged**